clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -69,6 +70,7 @@ public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V
         if (timestamp != null && timestamp < 0)
             throw new IllegalArgumentException(
                     String.format("Invalid timestamp: %d. Timestamp should always be non-negative or null.", timestamp));
+        Topic.validate(topic);
```

Comment:
One issue is that the exception thrown by this method is `InvalidTopicException`. In this context, `IllegalArgumentException` seems like the right exception to throw to be consistent with the other things we check. Maybe we can catch the exception and rethrow it.

Also, it would be good to mention it in a @throws clause.